pkg/reconciler/taskrun/config.go

```diff
@@ -0,0 +1,391 @@
+		// 	}
+		// }
+
+		config.DynamicPoolPlatforms[platform] = poolConfig
+	}
+
+	// Parse static hosts (host.<name>.* pattern)
+	staticHosts := make(map[string]StaticHostConfig)
+	for key, value := range data {
+		if !strings.HasPrefix(key, "host.") {
+			continue
+		}
+
+		// Extract host name and field from key
+		key = key[len("host."):]
+		pos := strings.LastIndex(key, ".")
+		if pos == -1 {
+			continue
+		}
+
+		hostName := key[0:pos]
+		field := key[pos+1:]
+
+		// Get or create host config
+		hostConfig, exists := staticHosts[hostName]
+		if !exists {
+			hostConfig = StaticHostConfig{}
+		}
+
+		// Parse and validate each field
+		switch field {
+		case "address":
+			if err := validateIP(value); err != nil {
+				return nil, fmt.Errorf("static host '%s': invalid address '%s': %w", hostName, value, err)
+			}
+			hostConfig.Address = value
+
+		case "user":
+			if strings.TrimSpace(value) == "" {
+				return nil, fmt.Errorf("static host '%s': user field cannot be empty", hostName)
+			}
+			hostConfig.User = value
+
+		case "platform":
+			if err := validatePlatformFormat(value); err != nil {
+				return nil, fmt.Errorf("static host '%s': invalid platform '%s': %w", hostName, value, err)
+			}
+			hostConfig.Platform = value
+
+		case "secret":
+			// For IBM platforms, validate using validateIBMHostSecret
+			if strings.Contains(hostConfig.Platform, "s390x") || strings.Contains(hostConfig.Platform, "ppc64le") {
+				if err := validateIBMHostSecret(hostConfig.Platform, value); err != nil {
+					return nil, fmt.Errorf("static host '%s': invalid secret '%s': %w", hostName, value, err)
+				}
+			}
+			hostConfig.Secret = value
+
+		case "concurrency":
+			concurrency, err := validateNumericValue(value, maxStaticConcurrency)
+			if err != nil {
+				return nil, fmt.Errorf("static host '%s': invalid concurrency '%s': %w", hostName, value, err)
+			}
+			hostConfig.Concurrency = concurrency
+		}
+
+		staticHosts[hostName] = hostConfig
+	}
+
+	// Add validated static hosts to config, keyed by hostname
+	for hostName, hostConfig := range staticHosts {
+		// Only validate that address field was provided (others are validated during parsing)
+		if hostConfig.Address == "" {
+			return nil, fmt.Errorf("static host '%s': address field is required", hostName)
+		}
+
+		config.StaticPlatforms[hostName] = hostConfig
+	}
+
+	// Parse default instance tag
+	if instanceTag := data[DefaultInstanceTag]; instanceTag != "" {
+		config.DefaultInstanceTag = instanceTag
+	}
+
+	// Parse additional instance tags
+	if additionalTags := data[AdditionalInstanceTags]; additionalTags != "" {
+		tags := strings.Split(additionalTags, ",")
+		for _, tag := range tags {
+			tag = strings.TrimSpace(tag)
+			parts := strings.Split(tag, "=")
+			if len(parts) >= 2 {
+				config.AdditionalInstanceTags[parts[0]] = parts[1]
+			} else {
+				return nil, fmt.Errorf("invalid additional instance tag format '%s': must be key=value", tag)
+			}
```

Comment:
if more than one `=` is found, the format is still invalid. We should return an error.

```suggestion
			if len(parts) != 2 {
				return nil, fmt.Errorf("invalid additional instance tag format '%s': must be key=value", tag)
			}
			config.AdditionalInstanceTags[parts[0]] = parts[1]
```